testing/assertions_skia.h

```diff
@@ -7,15 +7,57 @@
 
 #include <ostream>
 
+#include "third_party/skia/include/core/SkClipOp.h"
 #include "third_party/skia/include/core/SkMatrix.h"
 #include "third_party/skia/include/core/SkMatrix44.h"
+#include "third_party/skia/include/core/SkPaint.h"
+#include "third_party/skia/include/core/SkPath.h"
 #include "third_party/skia/include/core/SkPoint3.h"
 #include "third_party/skia/include/core/SkRRect.h"
 
 //------------------------------------------------------------------------------
 // Printing
 //------------------------------------------------------------------------------
 
+inline std::ostream& operator<<(std::ostream& os, const SkClipOp& o) {
+  switch (o) {
+    case SkClipOp::kDifference:
+      os << "ClipOpDifference";
+      break;
+    case SkClipOp::kIntersect:
+      os << "ClipOpIntersect";
+      break;
+#ifdef SK_SUPPORT_DEPRECATED_CLIPOPS
+    case SkClipOp::kUnion_deprecated:
+      os << "ClipOpUnion_deprecated";
+      break;
+    case SkClipOp::kXOR_deprecated:
+      os << "ClipOpXOR_deprecated";
+      break;
+    case SkClipOp::kReverseDifference_deprecated:
+      os << "ClipOpReverseDifference_deprecated";
+      break;
+    case SkClipOp::kReplace_deprecated:
+      os << "ClipOpReplace_deprectaed";
+      break;
+#else
+    case SkClipOp::kExtraEnumNeedInternallyPleaseIgnoreWillGoAway2:
+      os << "ClipOpReserved2";
+      break;
+    case SkClipOp::kExtraEnumNeedInternallyPleaseIgnoreWillGoAway3:
+      os << "ClipOpReserved3";
+      break;
+    case SkClipOp::kExtraEnumNeedInternallyPleaseIgnoreWillGoAway4:
+      os << "ClipOpReserved4";
+      break;
+    case SkClipOp::kExtraEnumNeedInternallyPleaseIgnoreWillGoAway5:
+      os << "ClipOpReserved5";
+      break;
+#endif
+  }
+  return os;
+}
+
 inline std::ostream& operator<<(std::ostream& os, const SkMatrix& m) {
```

Comment:
@chinmaygarde : do you remember why you're using `inline` here? I think you're usually most against using `inline`...

Reply:
`inline` is necessary here because importing this header in multiple TU will lead to duplicate symbols.

Reply:
The other option is to explicitly extern this and in a `.cc` file and add the implementation there. But for a test, I suppose this is fine. But yeah, we have a pretty strong bias against inlining unless instrumented to be necessary.

Reply:
I'll make an issue/PR just for you guys and do the extern'ing (in a follow-on CL)